src/NosCore.Database/DAL/ItemInstanceDAO.cs

```diff
@@ -35,250 +38,267 @@
 
 namespace NosCore.Database
 {
-    public class ItemInstanceDao : IGenericDao<IItemInstanceDto>
+    public class ItemInstanceDao : Dao<ItemInstance, IItemInstanceDto, Guid>
     {
         private readonly ILogger _logger;
-        private readonly PropertyInfo? _primaryKey;
+        private readonly PropertyInfo[] _primaryKey;
+        private readonly IDbContextBuilder _dbContextBuilder;
 
-        public ItemInstanceDao(ILogger logger)
+        public ItemInstanceDao(ILogger logger, IDbContextBuilder dbContextBuilder) : base(logger, dbContextBuilder)
         {
             _logger = logger;
-            try
-            {
-                _primaryKey = typeof(IItemInstanceDto).FindKey();
-                if (_primaryKey != null)
-                {
-                    return;
-                }
-
-                throw new KeyNotFoundException();
-            }
-            catch (Exception e)
-            {
-                _logger.Error(e.Message, e);
-            }
+            _dbContextBuilder = dbContextBuilder;
+            using var context = _dbContextBuilder.CreateContext();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'context'.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5326532)